### PR TITLE
DAOS-5803 example: add simple example for different object APIs

### PR DESCRIPTION
**Pull request by mchaarawi**
Skip-func-test-vm: true
Skip-func-hw-test-small: true
Skip-func-hw-test-medium: true
Skip-func-hw-test-large: true

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

### CODE REVIEW

**Review by jolivier23 — DISMISSED** [review]
should we actually run the example ?  I've done example code in the past for other projects and found it gets stale pretty fast.

**Comment by mchaarawi** [comment]
> should we actually run the example ? I've done example code in the past for other projects and found it gets stale pretty fast.

agreed, but i just wanted something to share with others for now.

**Review by jolivier23 — APPROVED** [review]

File: src/tests/simple_obj.c [hunk 1]
```diff
@@ -0,0 +1,782 @@
...
+	char		akey_str[10];
+	int		i, rc;
+
+	MPI_Barrier(MPI_COMM_WORLD);
+	if (rank == 0)
+		printf("Example of DAOS Key Single Value type:\n");
+
+	/*
+	 * Most of this example is the same as the key_array one, except the
+	 * type of the value under tha akey will be a single value of size
```
Comment (jolivier23) [inline src/tests/simple_obj.c:355]: 11:19:09  src/tests/simple_obj.c:355: tha ==> than, that, the

This is what is causing checkpatch to fail

**Review by jolivier23 — CHANGES_REQUESTED** [review]

File: src/tests/simple_obj.c (hunk 1 above)
Comment (jolivier23) [reply src/tests/simple_obj.c:355]: There is an open bug on this but it happens when the suggestion is more than one word

**Review by jolivier23 — APPROVED** [review]

**Review by johannlombardi — APPROVED** [review]

**Pull request merged**